app/cmd/runner.go

```diff
@@ -0,0 +1,133 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/jenkins-zh/jenkins-cli/app/cmd/common"
+	"github.com/jenkins-zh/jenkins-cli/app/i18n"
+	"github.com/jenkins-zh/jenkins-cli/util"
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+// RunnerOption is the wrapper of jenkinsfile runner cli
+type RunnerOption struct {
+	common.BatchOption
+	common.CommonOption
+	RoundTripper http.RoundTripper
+
+	Safe            bool
+	WarVersion      string
+	WarPath         string
+	PluginPath      string
+	JenkinsfilePath string
+	JfrVersion      string
+	LTS             bool
+}
+
+var runnerOption RunnerOption
+
+func init() {
+	rootCmd.AddCommand(runnerCmd)
+	runnerOption.SetFlag(runnerCmd)
+	runnerCmd.Flags().StringVarP(&runnerOption.WarPath, "path", "w", "",
+		i18n.T("The jenkins.war path"))
+	runnerCmd.Flags().BoolVarP(&runnerOption.Safe, "safe", "s", true,
+		i18n.T("Puts Jenkins into the quiet mode, wait for existing builds to be completed, and then restart Jenkins"))
+	runnerCmd.Flags().StringVarP(&runnerOption.WarVersion, "war-version", "v", "2.190.3",
+		i18n.T("The of version of jenkins.war"))
+	runnerCmd.Flags().StringVarP(&runnerOption.PluginPath, "plugin-path", "p", "",
+		i18n.T("The path to plugins.txt"))
+	runnerCmd.Flags().StringVarP(&runnerOption.JenkinsfilePath, "jenkinsfile-path", "z", "",
+		i18n.T("The path to jenkinsfile"))
+	runnerCmd.Flags().StringVarP(&runnerOption.JfrVersion, "jfr-version", "f", "1.0-beta-11",
+		i18n.T("The path to jenkinsfile"))
+	runnerCmd.Flags().BoolVarP(&runnerOption.LTS, "lts", "", true,
+		i18n.T("If you want to download Jenkins as LTS"))
+}
+
+var runnerCmd = &cobra.Command{
+	Use:   "runner",
+	Short: i18n.T("The wrapper of jenkinsfile runner"),
+	Long: i18n.T(`The wrapper of jenkinsfile runner
+Get more about jenkinsfile runner from https://github.com/jenkinsci/jenkinsfile-runner`),
+	RunE: func(cmd *cobra.Command, _ []string) (err error) {
+
+		var userHome string
+		if userHome, err = homedir.Dir(); err != nil {
+			return err
+		}
+		//Start by downloading the mirror for the jenkinsfileRunner
+		jenkinsfileRunnerVersion := fmt.Sprintf("jenkinsfile-runner-%s.jar", runnerOption.JfrVersion)
+		jenkinsfileRunnerTargetPath := fmt.Sprintf("%s/.jenkins-cli/cache/%s/%s", userHome, runnerOption.WarVersion, jenkinsfileRunnerVersion)
+		jenkinsfileRunnerURL := fmt.Sprintf("https://repo.jenkins-ci.org/list/releases/io/jenkins/jenkinsfile-runner/jenkinsfile-runner/1.0-beta-11/%s", jenkinsfileRunnerVersion)
+		logger.Info("Prepare to start Downloading jenkinfileRunner", zap.String("URL", jenkinsfileRunnerURL))
+		downloader := util.HTTPDownloader{
+			URL:            jenkinsfileRunnerURL,
+			ShowProgress:   true,
+			TargetFilePath: jenkinsfileRunnerTargetPath,
+		}
+		if err := downloader.DownloadFile(); err != nil {
```

Comment:
always download this file might not be necessary

Reply:
Yes should I check instead if the file is already present ?

Reply:
Yes, you can give it an option whether the user wants to overwrite it. Plus, the default value should be false due to the safety reason.

Reply:
Done :)